assets/js/admin-settings/index.jsx

```diff
@@ -0,0 +1,118 @@
+/* global localizedData */
+import {
+  Button,
+  SelectControl,
+  TextControl,
+  ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect, useState } from 'react';
+import useSiteOptions from '../services/hooks/use-site-options';
+import Rule from './rule';
+import { ruleCorral } from './styles';
+
+
+const AdminSettings = () => {
+  const [{ loading, saving, settings }, setOptions] = useSiteOptions();
+  const [originIndex, setOriginIndex] = useState(null);
+  const [targetIndex, setTargetIndex] = useState(null);
+
+  /**
+   * Helper function for saving the in-memory settings to the server.
+   */
+   const saveSettings = (updatedRules) => {
+    const next = { ...settings, apple_news_automation: updatedRules };
+
+    // Enforce some defaults prior to save.
+    // Request will 500 when site_logo === null.
+    next.site_logo = next.site_logo ?? 0;
+    console.log('settings-pre-save', next);
+
+    // Kick off the save to the server.
+    setOptions(next);
+  };
+  
+  const cleanData = () => {
+    saveSettings([]);
+  }
+  
+  // useEffect(() => {
+  //   cleanData();
+  // },[]);
+
+  const addRule = (newRule) => {
```

Comment:
Add should just be adding an "empty" rule to the end of the list, and then the normal loop/edit/update functionality can take over.